lib/JITServer/JITServer.cpp

```diff
@@ -218,28 +257,23 @@ ServerInitializeThreadContext(
     {
         return hr;
     }
-#ifdef USE_RPC_HANDLE_MARSHALLING
-    HANDLE targetHandle;
-    if (!DuplicateHandle(GetCurrentProcess(), processHandle, GetCurrentProcess(), &targetHandle, 0, false, DUPLICATE_SAME_ACCESS))
-    {
-        return E_ACCESSDENIED;
-    }
-#else
-    HANDLE targetHandle = OpenProcess(PROCESS_VM_OPERATION | PROCESS_VM_READ | PROCESS_VM_WRITE | PROCESS_QUERY_LIMITED_INFORMATION, false, clientPid);
-    if (!targetHandle)
+    ProcessContext* processContext = ProcessContextManager::GetProcessContext(clientPid);
+    if (processContext == nullptr)
     {
         return E_ACCESSDENIED;
     }
-#endif
     try
     {
         AUTO_NESTED_HANDLED_EXCEPTION_TYPE(static_cast<ExceptionType>(ExceptionType_OutOfMemory));
-        contextInfo = HeapNew(ServerThreadContext, threadContextData, targetHandle);
+        contextInfo = HeapNew(ServerThreadContext, threadContextData, processContext);
         ServerContextManager::RegisterThreadContext(contextInfo);
     }
     catch (Js::OutOfMemoryException)
     {
-        CloseHandle(targetHandle);
+        if (!contextInfo)
```

Comment:
>contextInfo [](start = 13, length = 11)

if there's OOM after HeapNew, we should free contextInfo
if it's not possible to OOM after HeapNew, the we don't need the check here